index.js

```diff
@@ -41,21 +67,20 @@ async function viaContentsApi({
 // Pros: one request + maybe doesn't require token
 // Cons: huge on huge repos + may be truncated
 async function viaTreesApi({
-	user,
-	repository,
-	ref = 'HEAD',
-	directory,
+	resource,
 	token,
 	getFullData = false
 }) {
-	if (!directory.endsWith('/')) {
-		directory += '/';
+	const res = parseResource(resource);
+
+	if (!res.directory.endsWith('/')) {
+		resource.directory += '/';
```

Comment:
I think this was a mistake, it should be moved after the API call, or else the API endpoint becomes `user/repo//git/trees/ref?recursive=1`, notice the double `/` after repo

Reply:
Fixed!